content/docs/04-clusters/03-edge/02-edge-configuration/30-cloud-init.md

```diff
@@ -59,6 +59,63 @@ You may ask yourself where to use cloud-init stages, as both the Edge Installer
 
+# Sensitive Information in the User Data Stages
+
+Suppose you plan to use sensitive information in your user data stages. In the edge deployment lifecycle, you can apply user data to edge hosts during the staging and site installation steps, as highlighted in the diagram below.  
+
+![A diagram highlighting the two stages in the edge deployment lifecycle where you can apply user data.](/edge_edge-configuration_cloud-init_user-data.png)
+<br />
+
+Generally, the user data is copied to the edge hosts in the above-mentioned steps, described briefly in the points below. 
+<br />
+
+- **Staging Step** - In the staging step, you prepare your edge hosts using the organization-level configurations. The organization-level configurations include the Edge Installer, the user data, and, optionally, a content bundle. 
+You bundle the organization-level configurations into an ISO and prepare a bootable USB flash drive. You use the USB flash drive to boot up the edge host and install the configurations. The entire USB flash drive data is copied to the edge host during the installation. 
+
+  Once the edge host is prepared with the initial installation, you remove the USB flash drive and ship your devices to the site for installation. This step is also called the *installer handoff* step. Refer to the [Prepare Edge Host
+](/clusters/edge/site-deployment/stage#prepareedgehost) guide to learn more about driving the installer handoff step.
+
+
+- **Site Installation Step** - In the site installation step, you use supplementary user data to apply site-specific configurations to the edge devices. In this step, you create another USB flash drive using an ISO that contains your additional user data. You use the USB flash drive to install the site-specific configurations. The entire USB flash drive data is copied to the edge host during the installation unless you follow the specific naming convention for your user data stages as described below. 
+
+  The [Perform Site Install](/clusters/edge/site-deployment/site-installation) guide describes the site installation process in detail. The site installation step is optional. You can refer to the [Multiple User Data Use Case](/clusters/edge/edgeforge-workflow/prepare-user-data#multipleuserdatausecase) guide to learn about the use cases for the site installation step.
+
+
+In both steps mentioned above,  the Edge Installer copies the user data from the USB flash drive to the **/run/stylus/userdata** file or the **/oem/userdata** file on the edge hosts. Suppose you want to keep some user data stages off the edge hosts. In such scenarios, we recommend applying the user data during the site installation step and using the specific naming convention for your user data stages, as described in the [Sensitive Information During the Site Installation](#sensitiveinformationduringthesiteinstallation) section below.
```

Comment:
```suggestion
In both steps mentioned above,  the Edge Installer copies the user data configuration file provided to the **/run/stylus/userdata** file or the **/oem/userdata** file on the edge hosts. If you want to prevent some user data stages from getting copied to the edge host's storage, you can use a specific naming convention to disable the default copy behavior. However, be aware that different persistence behaviors apply depending on which stage of the Edge deployment life cycle you provide sensitive data in the user data configuration file. Refer to the [Sensitive Information During the Site Installation](#sensitiveinformationduringthesiteinstallation) section below to learn more.
```

This paragraph was phrased awkwardly. It started explaining the use case then it transitioned into a recommendation and then back to the use case and how.  I also removed the USB disk in the beginning because that don't always apply to the fist stage, as they could be using a PXE server to load the device.